src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -9,6 +9,53 @@
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RegistrationException("User can`t be null!");
+        }
+
+        if (user.getLogin() == null || user.getLogin().length() < 6) {
+            throw new RegistrationException("Login must contain at least 6 symbols!");
+        }
+
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RegistrationException("User with this login already exists!");
+        }
+
+        if (user.getPassword() == null || user.getPassword().length() < 6) {
```

Comment:
You're checking if the password is null or less than 6 characters long. This is good, but according to the checklist, you should also test edge cases, such as a password of exactly 6 characters.